File: app/src/main/java/deeplinking/com/mvvmsample/adapter/NewsRecyclerAdapter.java
package deeplinking.com.mvvmsample.adapter;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.ArrayList;

import deeplinking.com.mvvmsample.R;
import deeplinking.com.mvvmsample.databinding.ListRowBinding;
import deeplinking.com.mvvmsample.viewmodel.NewsModel;

/**
 * Created by Rajesh Kumar on 10-05-2018.
 */
public class NewsRecyclerAdapter extends RecyclerView.Adapter<NewsRecyclerAdapter.ViewHolder> {


    Context context;
    ArrayList<NewsModel> arrayList;

    public NewsRecyclerAdapter(Context context, ArrayList<NewsModel> arrayList) {
        this.context = context;
        this.arrayList = arrayList;
    }


    class ViewHolder extends RecyclerView.ViewHolder {
        private final ListRowBinding rowBinding;

        public ViewHolder(ListRowBinding itemView) {
            super(itemView.getRoot());
            this.rowBinding = itemView;
        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        ListRowBinding itemBinding = DataBindingUtil.inflate(layoutInflater, R.layout.list_row, parent, false);
        return new ViewHolder(itemBinding);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

        holder.rowBinding.setNewslist(arrayList.get(position));

    }


    @Override
    public int getItemCount() {
        return arrayList.size();
    }
}

File: app/src/main/java/deeplinking/com/mvvmsample/adapter/NewsRecyclerAdapter.kt
package deeplinking.com.mvvmsample.adapter

import android.content.Context
import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import java.util.ArrayList

import deeplinking.com.mvvmsample.R
import deeplinking.com.mvvmsample.databinding.ListRowBinding
import deeplinking.com.mvvmsample.viewmodel.NewsModel

/**
 * Created by Rajesh Kumar on 10-05-2018.
 */
class NewsRecyclerAdapter(internal var context: Context, internal var arrayList: ArrayList<NewsModel>) : RecyclerView.Adapter<NewsRecyclerAdapter.ViewHolder>() {


    inner class ViewHolder( val rowBinding: ListRowBinding) : RecyclerView.ViewHolder(rowBinding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val itemBinding = DataBindingUtil.inflate<ListRowBinding>(layoutInflater, R.layout.list_row, parent, false)
        return ViewHolder(itemBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        holder.rowBinding!!.newslist = arrayList[position]

    }


    override fun getItemCount(): Int {
        return arrayList.size
    }
}
